packages/phalanx/test/Field/MDField_Compiletime.cpp

```diff
@@ -417,6 +417,18 @@ TEUCHOS_UNIT_TEST(mdfield, CompileTimeChecked)
       c_f7.setFieldData(PHX::KokkosViewFactory<double,typename PHX::DevLayout<double>::type,PHX::Device>::buildView(c_f7.fieldTag()));
     }
 
+    // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    // Kokkos cast DynRankView accessor
+    {
+      auto kva = a.get_view();
+      Kokkos::deep_copy(kva, a.get_view());
+
+      auto kva_s = a.get_static_view();
+      Kokkos::deep_copy(kva_s, a.get_view());
+
+      auto kvc = c.get_view();
+      Kokkos::deep_copy(kvc, c.get_static_view());
+    }
     // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
A silly question - the tests are doing a deep_copy to itself. Are we sure that kokkos doesn't look at the pointer and skip the operations knowing they are the same memory? Just wondering if these tests should use a host mirror to make sure a copy really takes place?

Reply:
That is correct, before that code would have compile time errored out.  This is saying that the compiler can now do the cast and call the deep copy function.  What happens inside is not part of the test.  I can do a deep copy with a const and check that const,

Reply:
Crap, it isn't doing what I want it to, let me dig more

Reply:
probably too hopped up on cold meds to be doing this :). I will look into what I was thinking but it wasn't testing any of the new code...